bliss.js

```diff
@@ -363,12 +363,17 @@ extend($, {
 			}
 		}
 
-		if (env.method !== "GET" && !env.headers["Content-type"] && !env.headers["Content-Type"]) {
+		var hasContentType = Boolean(Object.keys(env.headers).find(function(key) {
```

Comment:
No need to call `Boolean()`, a) `!!` is shorter and b) conditionals work with truthy/falsy values, not necessarily booleans.

Also `Object.keys(env.headers).map(function(key) { return key.toLowerCase() }).indexOf("content-type")` might work better, but that's up to you

Also are you absolutely sure that there's no possible value we can set later to unset this, so that we don't have to check? Have you tried `""`? `null`?.

Reply:
There are no mistakes here. Just personal preference.

> `!!` is shorter

`!!` is actually two separate `!` operators and hence a bit of a "hack", and obviously slower (huge relative difference, but very small real world difference). It's considered bad practice by many (including me): https://eslint.org/docs/rules/no-implicit-coercion. And it doesn't really save you a lot of characters.

> conditionals work with truthy/falsy values

You're right that the type coercion `Boolean()` is not providing any functional value. That's not what it's for. `if (Boolean(x)) { ... }` or `if (!!x) { ... }` are both using using redundant type coercion. But I'm assigning to a variable `hasContentType`. That's a required intermediate step to avoid the `if` statement growing to multiple lines and getting really hard to read.

For readability `hasContentType` (boolean) is a better name than `ContentTypeCasing` (string), the latter being what `find()` returns if there is a match and you don't force-coerce it.

The alternative (that I think you're suggesting?) is obscuring the type (pretending it's a boolean). I don't like that, and think it's more likely to cause confusion later on. Semantics matter.

> Also are you absolutely sure that there's no possible value we can set later to unset this, so that we don't have to check? Have you tried ""? null?.

Yes. All values except `undefined` will be set in the request header, including null, '', and 0 (although null will be converted to `"null"`).

There is no type coercion with `!==`.

> `Object.keys(env.headers).map(function(key) { return key.toLowerCase() }).indexOf("content-type")`

`Array.find()` stops after the first match, and hence avoid operations needed by `Array.map` + `indexOf`. Performance won't be a problem either way here, but out of the two I do think `Array.find()` is better (unless I'm missing something?).

This isn't important to me though, and if I go with your suggestion I could make the intermediate variable `headerKeys` (lowercased) instead of `hasContentType`, and therefore wouldn't need the coercion. Does that sound good?

Reply:
I think I like this better, because if we run into any future issues where we need to check headers, we already have `headerKeys`.

Reply:
Also, super minor nitpick: blank line before control flow statements please (I should add an ESLint rule enforcing this at some point, sorry I haven't yet!)

Reply:
Updated! I also added that eslint-rule as a separate PR #206 (required eslint upgrade)